homeassistant/helpers/device_registry.py

```diff
@@ -1051,7 +1077,24 @@ def async_cleanup(
 @callback
 def async_setup_cleanup(hass: HomeAssistant, dev_reg: DeviceRegistry) -> None:
     """Clean up device registry when entities removed."""
-    from . import entity_registry  # pylint: disable=import-outside-toplevel
+    # pylint: disable-next=import-outside-toplevel
+    from . import entity_registry, label_registry as lr
+
+    @callback
+    def _label_removed_from_registry_filter(event: Event) -> bool:
+        """Filter all except for the remove action from label registry events."""
+        return bool(event.data["action"] == "remove")
```

Comment:
Why bool it?

Reply:
Yeah odd huh? Data can be any, thus it could be something that implements the equality subset method, which doesn't have to return a boolean.

Long story short: mypy wants it.

Reply:
The problem is, that async_listen will not accept that either.

Let's assume this:

![CleanShot 2024-02-18 at 10 31 51@2x](https://github.com/home-assistant/core/assets/195327/dfadf74d-bc63-451b-8a12-35a0bc7272f7)

This will result in:

```
homeassistant/helpers/device_registry.py:1099: error: Argument "event_filter" to "async_listen" of "EventBus" has incompatible type "Callable[[EventType[EventLabelRegistryUpdatedData]], bool]"; expected "Callable[[Event], bool] | None"  [arg-type]
homeassistant/helpers/device_registry.py:1100: error: Argument "listener" to "async_listen" of "EventBus" has incompatible type "Callable[[EventType[EventLabelRegistryUpdatedData]], None]"; expected "Callable[[Event], Coroutine[Any, Any, None] | None]"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```

How that is solved on some places, is by silencing the arg-type rule:

![CleanShot 2024-02-18 at 10 33 44@2x](https://github.com/home-assistant/core/assets/195327/65a79b5f-3a26-4102-9800-18b6bf0796b8)

As far as I understood, we can't turn `Event` into a generic just of yet.

Not sure what we prefer here. Since I've found some occurances, I'll go for ignoring the arg-type rule 🤷

Reply:
Done in [9c7748b](https://github.com/home-assistant/core/pull/110821/commits/9c7748bf88732f594a2273cafd433c64b6466a75)